### PR TITLE
Fixes #3385 - groups with same IDs as apps should not be created.

### LINKED ISSUE #3385: Creating an empty group using an existing app ID should return 409
When sending a `POST` to `/v2/groups` with a group `{"id": "/group/existing-appname"}` I’d expect to receive a `409`, but I am getting a `200` instead 

```
{
  deploymentId: "7a8ad237-a76b-428e-8915-5dc9c98716b3"
  version: "2016-03-02T11:02:45.586Z"
}
```

### PR DESCRIPTION
**Pull request by alexanderweber**
Fixes #3385.

### CODE REVIEW

**Comment by alexanderweber** [comment]
@aquamatthias there was no check against identical path definition of an app with a newly created group. It was possible to create an empty group `"/group/existing-appname"` even if there was an app created as `"/group/existing-appname"` before.

I did add this check in GroupsResource:

```Scala
result(groupManager.rootGroup()).transitiveApps.find(_.id == path)
```

Please merge, if you agree with the change.

File: src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala [hunk 1]
```diff
@@ -105,14 +105,18 @@ class GroupsResource @Inject() (
...
 
-      result(groupManager.rootGroup()).findGroup(_.id == effectivePath) match {
+      result(groupManager.rootGroup()).findGroup(_.id == path) match {
         case Some(currentGroup) =>
-          throw ConflictingChangeException(s"Group $effectivePath is already created. Use PUT to change this group.")
+          throw ConflictingChangeException(s"Group $path is already created. Use PUT to change this group.")
         case None =>
-          val (deployment, path) = updateOrCreate(id.toRootPath, groupUpdate, force)
-          deploymentResult(deployment, Response.created(new URI(path.toString)))
+          result(groupManager.rootGroup()).transitiveApps.find(_.id == path) match {
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala]: The call to groupManager.rootGroup() is expensive and you call it twice here. Please reuse the root group from the first call.

Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala]: Also: nested match elements are sometimes hard to read: extract def's

**Comment by aquamatthias** [comment]
@alexanderweber thanks - added some suggestions.

File: src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala (hunk 1 above)
Comment (alexanderweber) [inline src/main/scala/mesosphere/marathon/api/v2/GroupsResource.scala]: Totally agree. Fixed and encapsulated, let me know if it is ok.

**Comment by aquamatthias** [comment]
Looks great! Can we have a test for this?

**Comment by alexanderweber** [comment]
@aquamatthias I did write two tests. Let me know if those are ok.

**Comment by aquamatthias** [comment]
@thanks!

**Pull request merged**